extensions/functions_datetime.yaml

```diff
@@ -6,25 +6,49 @@ scalar_functions:
     description: Extract portion of a date/time value.
     impls:
       - args:
-          - options: [ YEAR, MONTH, DAY, SECOND ]
+          - options: [ YEAR, ISO_YEAR, QUARTER, MONTH, DAY, DAY_OF_WEEK, ISO_DAY_OF_WEEK, DAY_OF_YEAR, WEEK,
+                       ISO_WEEK, HOUR, MINUTE, SECOND, MILLISECOND, MICROSECOND, NANOSECOND, SUBSECOND, EPOCH ]
             name: The part of the value to extract.
             required: true
           - value: timestamp
         return: i64
       - args:
-          - options: [ YEAR, MONTH, DAY, SECOND ]
+          - options: [ YEAR, ISO_YEAR, QUARTER, MONTH, DAY, DAY_OF_WEEK, ISO_DAY_OF_WEEK, DAY_OF_YEAR, WEEK,
+                       ISO_WEEK, HOUR, MINUTE, SECOND, MILLISECOND, MICROSECOND, NANOSECOND, SUBSECOND, EPOCH,
+                       TIMEZONE_OFFSET ]
             name: The part of the value to extract.
             required: true
           - value: timestamp_tz
+          - value: string
         return: i64
       - args:
-          - options: [ YEAR, MONTH, DAY ]
+          - options: [ IS_LEAP_YEAR ]
+            name: The part of the value to extract.
+            required: true
+          - value: timestamp
+        return: boolean
+      - args:
+          - options: [ IS_LEAP_YEAR, IS_DST ]
+            name: The part of the value to extract.
+            required: true
+          - value: timestamp_tz
+          - value: string
+        return: boolean
```

Comment:
This has to be a separate function, or it should return 0/1 as an i64 like the others. Substrait doesn't support differentiating between implementations based upon only an option value, because:

> It is required that two function implementation with the same simple name must resolve to different compound names using types. If two function implementations in a YAML file resolve to the same compound name, the YAML file is invalid and behavior is undefined.

https://substrait.io/extensions/#function-signature-compound-names

Reply:
Moved to a separate function for now.